dropbox/dropbox.py

```diff
@@ -452,6 +466,39 @@ def request_json_string(self,
         else:
             raise HttpError(request_id, r.status_code, r.text)
 
+    def with_path_root(self, path_root=None):
+        """
+        Creates a new object with the same parameters as this object, but also
+        sets the ``Dropbox-API-Path-Root` header to :param:`path_root`.
+
+        :param PathRoot path_root: the path root object to serialize and use for
+        the ``Dropbox-API-Path-Root`` header. (If ``None``, the header is
+        cleared.)
+        :return: a new instance of this class with the ``Dropbox-API-Path-Root``
+        header set.
+        """
+        headers = self._headers
+
+        if headers is not None:
+            headers = dict(headers)  # make a copy that we can manipulate
+            headers.pop(PATH_ROOT_HEADER)
+
+        if path_root is not None:
+            if headers is None:
+                headers = {}
+            headers[PATH_ROOT_HEADER] = stone_serializers.json_encode(PathRoot_validator, path_root)
+
```

Comment:
`if <var> is not None:` is redundant.  Just use `if var`.

The above can be simplified to:

```python
headers = dict(self._headers or {})
if path_root:
   headers[PATH_ROOT_HEADER] = stone_serializers.json_encode(PathRoot_validator, path_root)
elif PATH_ROOT_HEADER in headers:
  headers.pop(PATH_ROOT_HEADER

# alternatively, if you want to get really fancy, you can write the "if path_root:" logic as
headers.update({PATH_ROOT_HEADER, stone_serializers.json_encode(PathRoot_validator, path_root}) if PATH_ROOT_HEADER else (PATH_ROOT_HEADER in headers and headers.pop(PATH_ROOT_HEADER)
# though it's a little more difficult to mentally dissect.

Reply:
Refactored (and simplified) in `set_path_root_header`.